ui/appui-react/src/appui-react/childwindow/CopyStyles.ts

```diff
@@ -2,39 +2,34 @@
  * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
-/**
- * Copies the source CSS into the destination
- * @param targetDoc - target document
- * @param sourceDoc - source document
+
+/** Copies the CSS style sheets from source document into the target document.
  * @internal
  */
 export function copyStyles(
   targetDoc: Document,
   sourceDoc: Document = document
 ) {
-  const stylesheets = sourceDoc.adoptedStyleSheets
-    ? Array.from([...sourceDoc.styleSheets, ...sourceDoc.adoptedStyleSheets])
-    : Array.from(sourceDoc.styleSheets);
+  const styleSheets = Array.from(sourceDoc.styleSheets);
+  styleSheets.forEach(({ ownerNode }) => {
+    // Copy `link` and `style` elements.
+    if (!ownerNode) return;
+    const clonedNode = targetDoc.importNode(ownerNode, true);
+    targetDoc.head.appendChild(clonedNode);
+  });
 
-  stylesheets.forEach((stylesheet) => {
-    const css = stylesheet;
-    if (stylesheet.href) {
-      const newStyleElement = targetDoc.createElement("link");
-      newStyleElement.rel = "stylesheet";
-      newStyleElement.href = stylesheet.href;
-      targetDoc.head.appendChild(newStyleElement);
-    } else {
-      if (css && css.cssRules && css.cssRules.length > 0) {
-        const newStyleElement = targetDoc.createElement("style");
-        Array.from(css.cssRules).forEach((rule) => {
-          newStyleElement.appendChild(targetDoc.createTextNode(rule.cssText));
-        });
-        targetDoc.head.appendChild(newStyleElement);
-      }
-    }
+  const adoptedStyleSheets = Array.from(sourceDoc.adoptedStyleSheets ?? []);
+  adoptedStyleSheets.forEach((styleSheet) => {
+    // Adopted stylesheet have no ownerNode and can't be shared between multiple documents.
+    if (!targetDoc.defaultView) return;
+    const newStyleSheet = new targetDoc.defaultView.CSSStyleSheet();
+    Array.from(styleSheet.cssRules).forEach((rule) => {
+      newStyleSheet.insertRule(rule.cssText);
```

Comment:
[`insertRule`](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleSheet/insertRule) inserts the rule at the beginning of the stylesheet, so I believe this will reverse the order of all CSS rules, which can lead to incorrect output.

You could specify the index manually, but I would actually suggest using the `stringifyStyleSheet` helper from https://github.com/iTwin/appui/issues/893#issuecomment-2218036702. As a bonus, it's more readable and would be easier to modify in the future (teaser: iTwinUI could one day provide a pre-serialized version of its stylesheet to bypass `cssText` altogether).

Reply:
Good catch, fixed by passing in an index for now.